### PR TITLE
[NFC] Clarify comment on metatype casts on TypeCheckConstraints

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
As pointed by @tbkka in #32234, the comment is not coherent with the code. So this is just clarifying the comment to match what is happening.

### CODE REVIEW

File: lib/Sema/TypeCheckConstraints.cpp [hunk 1]
```diff
@@ -3706,9 +3706,9 @@ CheckedCastKind TypeChecker::typeCheckCheckedCast(Type fromType,
   else
     fromRequiresClass = fromType->mayHaveSuperclass();
   
-  // Casts between protocol metatypes only succeed if the type is existential
-  // or if it involves generic types because they may be protocol conformances
-  // we can't know at compile time.
+  // Casts between metatypes only succeed if none of the types are existentials
+  // or if one is an existential and the other is a generic type because they
```
Comment (xedin) [inline lib/Sema/TypeCheckConstraints.cpp:3710]: Nit: they - there; and since we changing it - `we can't know` -> unknown.

Comment (LucianoPAlmeida) [reply lib/Sema/TypeCheckConstraints.cpp]: Fixed! Thanks :)

**Review by xedin — APPROVED** [review]

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by LucianoPAlmeida** [comment]
Thanks @xedin :)

**Pull request merged**